rollingpin/frontends.py

```diff
@@ -160,6 +161,28 @@ def on_deploy_end(self):
         elapsed = time.time() - self.start_time
         print "*** elapsed time: %d seconds" % elapsed
 
+        # TODO: Make this smarter.  It should only show anomolous
```

Comment:
Have you considered breaking the following into its own method? This method is getting really really long and unreadable. Also that way we easily can add tests to only test this portion of the code